hack/build-with-blog.sh

```diff
@@ -1,53 +1,111 @@
+rm -rf site/
+
+if [ "$BUILD_VERSIONS" == "no" ]; then
+  # HEAD to /docs if we're not doing versioning.
+  mkdocs build -f mkdocs.yml -d site/docs
+else
+  # Versioning: pre-release (HEAD): docs => development/
+  mkdocs build -f mkdocs.yml -d site/development
+
+  # Latest release branch to /docs
+  git clone --depth 1 -b ${BRANCHES[0]} https://github.com/${REPOS[0]}/docs "temp/docs-$latest"
+  pushd "temp/docs-$latest"
+  KNATIVE_VERSION=$latest mkdocs build -d ../../site/docs
+  popd
+
+  # Previous release branches release-$version to /v$version-docs
+  versionjson=""
+  for i in "${!previous[@]}"; do
+    version=${previous[$i]}
+    versionjson+="{\"version\": \"v$version-docs\", \"title\": \"v$version\", \"aliases\": [\"\"]},"
+    git clone --depth 1 -b ${BRANCHES[$i+1]} https://github.com/${REPOS[i+1]}/docs "temp/docs-$version"
+    pushd "temp/docs-$version"
+    KNATIVE_VERSION=$version VERSION_WARNING=true mkdocs build -d "../../site/v$version-docs"
+    popd
+  done
+
+  # Set up the version file to point to the built docs.
+  cat << EOF > site/versions.json
+  [
+    {"version": "docs", "title": "v$latest", "aliases": [""]},
+    $versionjson
+    {"version": "development", "title": "(Pre-release)", "aliases": [""]}
+  ]
+EOF
+fi
+
+if [ -z "$SKIP_BLOG" ]; then
+  # Clone out the website and community repos for the hugo bits.
+  # TODO(jz) Cache this and just do a pull/update/use siblings for local dev flow.
+  git clone --depth 1 https://github.com/knative/website temp/website
+  pushd temp/website; git submodule update --init --recursive --depth 1; popd
+  git clone -b ${community_branch} --depth 1 https://github.com/${community_repo}/community temp/community
+
+  # Move blog files into position
+  mkdir -p temp/website/content/en
+  cp -r blog temp/website/content/en/
+
+  # Clone community/ in to position too
+  # This is pretty weird: the base community is in docs, but then the
+  # community repo is overlayed into the community/contributing subdir.
+  cp -r community temp/website/content/en/
+  cp -r temp/community/* temp/website/content/en/community/contributing/
+  rm -r temp/website/content/en/community/contributing/elections/2021-TOC # Temp fix for markdown that confuses hugo.
+
+  # See https://github.com/knative/website/blob/main/scripts/processsourcefiles.sh#L125
+  # For the reasoning behind all this.
+  echo 'Converting all links in GitHub source files to Hugo supported relative links...'
+  pushd temp/website
+  # Convert relative links to support Hugo
+  find . -type f -path '*/content/*.md' ! -name '*_index.md' ! -name '*index.md' ! -name '*README.md' \
+    ! -name '*serving-api.md' ! -name '*eventing-contrib-api.md' ! -name '*eventing-api.md' \
+    ! -name '*build-api.md' ! -name '*.git*' ! -path '*/.github/*' ! -path '*/hack/*' \
+    ! -path '*/node_modules/*' ! -path '*/test/*' ! -path '*/themes/*' ! -path '*/vendor/*' \
+    -exec sed -i '/](/ { s#(\.\.\/#(../../#g; s#(\.\/#(../#g; }' {} +
```

Comment:
I guess we need the special  GNU `sed` on MacOS, same way as we have today in website repo build
I got an error
```
+ find . -type f -path '*/content/*.md' '!' -name '*_index.md' '!' -name '*index.md' '!' -name '*README.md' '!' -name '*serving-api.md' '!' -name '*eventing-contrib-api.md' '!' -name '*eventing-api.md' '!' -name '*build-api.md' '!' -name '*.git*' '!' -path '*/.github/*' '!' -path '*/hack/*' '!' -path '*/node_modules/*' '!' -path '*/test/*' '!' -path '*/themes/*' '!' -path '*/vendor/*' -exec sed -i '/](/ { s#(\.\.\/#(../../#g; s#(\.\/#(../#g; }' '{}' +
sed: 1: "./content/en/blog/relea ...": invalid command code .
```

Reply:
yeah, this is a copy paste from website repo (there's a couple links in community that need this while we're still building it with hugo), so same prereqs from there will be needed for this :(

Reply:
Can we add this to the script, and later in the MKDOCS.md dev guide @omerbensaadon is working on.

```
# 2. For Mac OSX: The script uses the `gnu` version of `sed`. To install `gnu-sed`, you use brew:#    1. Run `brew install gnu-sed`#    2. Add it to your `PATH`. For example, add the following line to your `~/.bash_profile`:#      `PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"`
```
This is the same we have in website repo https://github.com/knative/website/blob/9e5693d8cdf47af18ac1587be601b9e6c610d321/scripts/localbuild.sh#L13

Reply:
Might be able to do this in a cross-platform way (it is possible, I think, dont know why we didnt already), but I think we should probably land that separately

Reply:
yeah that sounds, good create an issue label mkdocs for the backlog to simply sed. for now just add the instructions in the script for folks to install with brew

Reply:
added a bit of docs to the top of the script for now, hopefully we can also create a docker container so folks dont have to think about this too much

Reply:
We already have a docker container @julz or you talking about something else?
https://github.com/knative/docs/blob/mkdocs/hack/docker/README.md

Reply:
I mean when we update it for this the install instructions should just be "run the container" iyswim